sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImplwithUGI.java

```diff
@@ -85,15 +73,6 @@ public String getDelegationToken() {
     return this.delegationTokenStr;
   }
 
-  @Override
-  protected synchronized void acquire(boolean userAccess) {
-    super.acquire(userAccess);
-    // if we have a metastore connection with impersonation, then set it first
-    if (sessionHive != null) {
-      Hive.set(sessionHive);
-    }
-  }
```

Comment:
I thought `HiveSessionImplwithUGI` need to override this because of the impersonation. Do you mean `HiveSessionImpl`'s implementation is enough to handle UGI use case?

Reply:
UGI is set by setSessionUGI？